src/primitive/phase_gate.jl

```diff
@@ -30,7 +30,6 @@ phase(0.1)
 ```
 """
 phase(θ::AbstractFloat) = PhaseGate(θ)
```

Comment:
```julia
julia> phase(π)
ERROR: MethodError: no method matching phase(::Irrational{:π})
Closest candidates are:
  phase(::AbstractFloat) at /home/leo/.julia/dev/YaoBlocks/src/primitive/phase_gate.jl:32
Stacktrace:
 [1] top-level scope at none:0
```

Reply:
at least support Irrational and integer like `0`? since shift does not error.
Note: `Basic` type is not a number or Real.

Reply:
I implemented my own symbolic system now, there are real scalar symbolic numbers, no worries.